Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -853,8 +813,8 @@ public class HTTPClient {
     public enum EventLoopGroupProvider {
         /// `EventLoopGroup` will be provided by the user. Owner of this group is responsible for its lifecycle.
         case shared(EventLoopGroup)
-        /// `EventLoopGroup` will be created by the client. When ``HTTPClient/syncShutdown()`` is called, the created `EventLoopGroup` will be shut down as well.
-        case createNew
+        /// An `EventLoopGroup` from `NIOSingletons`.
+        case globalSingleton
```

Comment:
Strictly speaking, yes. Practically I was hoping that outside clients only produce this value (which remains compatible because of the `static var createNew: Self { ... }`).

If yes want to follow SemVer to the tea then we should just start to offer `.globalSingleton` but leave the case as `.createNew`. I think we can still deprecate it.

Reply:
We can do this without breaking API by just deprecating `.createNew` and making `.globalSingleton` a static let that returns a `.shared(ELG)` as it does not need to be shutdown e.g.:
```swift
extension EventLoopGroupProvider {
    static let globalSingelton: Self = {
             #if canImport(Network)
             if #available(OSX 10.14, iOS 12.0, tvOS 12.0, watchOS 6.0, *) {
                 return .shared(NIOSingletons.multiThreadedTransportServicesEventLoopGroup)
             } else {
                 return .shared(NIOSingletons.multiThreadedPosixEventLoopGroup)
             }
             #else
             return .shared(NIOSingletons.multiThreadedPosixEventLoopGroup)
             #endif
    }()
}
```

Reply:
Yeah but I want `createNew` to return the singleton :). As described above, we can still deprecate `createNew` and create a new (non-deprecated) `static var globalSingleton` which does return `.createNew` but does tricks such that the deprecation warning doesn't show up. Not pretty but possible I think.

Reply:
As @FranzBusch said this is API breaking and therefore we can't do that.